src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -0,0 +1,1065 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Text.Json.Serialization.Metadata;
+using System.Text.Json.Tests;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class DefaultJsonTypeInfoResolverTests
+    {
+        [Fact]
+        public static void JsonPropertyInfoOptionsAreSet()
+        {
+            JsonSerializerOptions options = new();
+            JsonTypeInfo typeInfo = JsonTypeInfo.CreateJsonTypeInfo(typeof(MyClass), options);
+            CreatePropertyAndCheckOptions(options, typeInfo);
+
+            typeInfo = JsonTypeInfo.CreateJsonTypeInfo<MyClass>(options);
+            CreatePropertyAndCheckOptions(options, typeInfo);
+
+            typeInfo = options.TypeInfoResolver.GetTypeInfo(typeof(MyClass), options);
+            CreatePropertyAndCheckOptions(options, typeInfo);
+
+            static void CreatePropertyAndCheckOptions(JsonSerializerOptions expectedOptions, JsonTypeInfo typeInfo)
+            {
+                JsonPropertyInfo propertyInfo = typeInfo.CreateJsonPropertyInfo(typeof(string), "test");
```

Comment:
Do we have any tests validating the behavior of `CreateJsonPropertyInfo` for non-object kinds?